// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package helloworld;

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  
  // Ask for the user name
  rpc UserName (UserNameRequest) returns (UserNameReply) {}    
 
}

// The request hello message
message HelloRequest {
  string name = 1;
  string argument = 2;
}

// The response message containing the greetings (the initial hello)
message HelloReply {
  string message = 1;
}

// The response containing the username of the client
message UserNameReply {
  string message = 1; 
  string userName = 2; 
}

// The request message, asking for the users, users name
message UserNameRequest {
  string userName = 1; 
}

